tests/middleware/auth/test_auth_store.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from sqlalchemy.exc import IntegrityError, NoResultFound # type: ignore
+
+from src.data_access.models.user_model import User
+from src.middleware.auth.auth_store import AuthStore, IncorrectCredentialsError
+from src.middleware.exceptions.auth.user_already_exists_error import \
+    UserAlreadyExistsError
+from src.middleware.exceptions.auth.user_not_found_error import \
+    UserNotFoundError
+
+
+@pytest.fixture
+def mock_user(mocker):
+    mock_user = mocker.Mock()
+    mock_user.one.return_value = mock_user
+    mock_user.filter.return_value = mock_user
+    return mock_user
+
+
+@pytest.fixture
+def mock_db_session(mocker):
+    return mocker.Mock()
+
+
+@pytest.fixture
+def auth_store(mock_db_session):
+    return AuthStore(mock_db_session)
+
+
+class TestCreateUser:
+    def test_adds_new_user_to_db_session(self, mock_db_session, auth_store):
+        username, password = "1", "2"
+        user = User(username, password)
+        auth_store.create_user(user)
+        mock_db_session.add.assert_called_with(user)
+        mock_db_session.commit.assert_called_once()
```

Comment:
use assert_called_once_with so u sure it didnt get any parameter as u expect

Reply:
the commit method does not receive any parameters